bridgetown-core/lib/bridgetown-core/tags/class_map.rb

```diff
@@ -0,0 +1,90 @@
+# frozen_string_literal: true
+
+require "set"
+
+module Bridgetown
+  module Tags
+    # A ClassMap class is meant to take a hash and append styles based on if the
+    # value is truthy or falsy
+    #
+    # @example
+    #   center-var = true
+    #   small-var = nil
+    #
+    #   # input
+    #   <div class="{% class_map has-centered-text: center-var, is-small: small-var %}">
+    #     Text
+    #   </div>
+    #
+    #   # output
+    #   <div class="has-centered-text">
+    #     Text
+    #   </div>
+    class ClassMap < Liquid::Tag
+      # @see https://api.rubyonrails.org/classes/ActiveModel/Type/Boolean.html
+      FALSE_VALUES = [
+        nil, "nil", "NIL", false, 0, "0", :"0", "f", :f, "F", :F, "false",
+        false, "FALSE", :FALSE,
+      ].to_set.freeze
+
+      # @param tag_name [String] The name to use for the tag
+      # @param input [String] The input to the tag
+      # @param tokens [Hash] A hash of config tokens for Liquid.
+      #
+      #
+      # @return [ClassMap] Returns a ClassMap object
+      def initialize(tag_name, input, tokens)
+        super
+        @input = input
+      end
+
+      def render(context)
+        class_map(@input, context)
+      end
+
+      private
+
+      def class_map(string, context)
+        ary = []
+
+        string.split(%r!,\s+!).each do |item|
+          kv_pair = item.split(%r!:\s+!)
+          klass = kv_pair[0]
+          variable = kv_pair[1]
+
+          # Check if a user wants the opposite of the variable
+          if variable[0] == "!"
+            check_opposite = true
+            variable.slice!(1..-1)
+          end
+
+          variable = find_variable(context, variable)
+
+          if check_opposite
+            ary.push(klass) if FALSE_VALUES.include?(variable)
+          else
+            ary.push(klass) unless FALSE_VALUES.include?(variable)
+          end
+        end
+
+        ary.join(" ")
+
+      # Gracefully handle if syntax is improper
+      rescue NoMethodError
+        "invalid-class-map"
+      end
+
+      def find_variable(context, variable)
```

Comment:
Can we use the `lookup_variable` method from the `Bridgetown::LiquidExtensions` mixin? (TBH, I didn't really notice that was there until very recently! Should probably relocate the file to the `concerns` folder)

Reply:
I tried, but the issue is that it will return the string of the variable instead of nil. I need it to return nil so it can be omitted from output.

IE:

If the variables `foo` is nil, it will return `"foo"`

Reply:
[https://github.com/bridgetownrb/bridgetown/blob/main/bridgetown-core/lib/bridgetown-core/liquid_extensions.rb#L5-L20](https://github.com/bridgetownrb/bridgetown/blob/main/bridgetown-core/lib/bridgetown-core/liquid_extensions.rb#L5-L20)

Reply:
You know, I saw that `lookup || variable` bit and it threw me for a loop at first. I'm not even sure if that's desirable over the long term. Funny thing — I did a search and neither Bridgetown nor Jekyll seems to be using this in any production code. It seems it was added back in 2014 and nobody used it (maybe some third-party Jekyll plugin is using it, I dunno). I feel we could probably change the behavior to return nil…Liquid handles nil gracefully anyway and I simply can't imagine a case where anyone would *want* the variable name returned as a string.